HHS/simpler-grants-gov
# chore: update lockfile version for supported node version [2194]

## Summary
Fixes #2194

### Time to review: __15 mins__

## Changes proposed
Lockfile updated when running with up to date version of npm (10.8.2), in line with specified node version from package.json engines (>20).

## Context for reviewers

Lockfile version is updated from 1 to 3, with quite a large change to the formatting of the file. Somewhat explained here: https://nitayneeman.com/posts/catching-up-with-package-lockfile-changes-in-npm-v7/

Note that I tried to confirm that this change results in no changes to the installed node_modules, but when running an install on node 12 w/ npm version 6 in order to use the existing package lock format, the process errors out with the following, likely due to optional chaining not being supported in node 12:

```
/Users/dougschrashun/code/simpler-grants-gov/frontend/node_modules/tar-fs/index.js:6
const win32 = (global.Bare?.platform || process.platform) === 'win32'
                           ^

SyntaxError: Unexpected token '.'
```
I could try downgrading the tar-fs package to work around this but I feel like this is a safe enough operation to go through without that level of confirmation, and am unclear how the current package-lock is even still in place given the requirement on the engine version...

One other note - though the package.json specifies an engine of node > 20, it looks from this discussion like nextJS does not support > 18. With that in mind, though this change would be necessary to support node 18 in any case, should we pin at node 18 instead of 20?


### Test steps

1. `npm install`
2. _VERIFY_: no warnings or messages about outdated lockfile
3. `npm run dev`
4. _VERIFY_: application runs as expected


## Additional information
> Screenshots, GIF demos, code examples or output to help show the changes working as expected.
